### PR TITLE
MultiPointCollection to GeoJson

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

File: datatypes/src/collections/multi_point_collection.rs [hunk 1]
```diff
@@ -258,6 +259,134 @@ impl MultiPointCollection {
...
+    /// Helper function to create json values from the collection's multi points
+    fn coordinates_to_geo_json_geometries(&self) -> impl Iterator<Item = geojson::Geometry> + '_ {
+        let geometry_column: &ListArray = downcast_array(
+            &self
+                .data
+                .column_by_name(Self::FEATURE_COLUMN_NAME)
+                .expect("Column must exist, since it is in the metadata"),
+        );
+
+        (0..self.len()).map(move |i| {
```
Comment (jdroenner) [inline datatypes/src/collections/multi_point_collection.rs]: is there a way to have an iterator over the entries instead of the 0..len range?

Comment (ChristianBeilschmidt) [reply datatypes/src/collections/multi_point_collection.rs]: d387147

File: datatypes/src/primitives/multi_point.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+use crate::error;
+use crate::primitives::Coordinate2D;
+use crate::util::Result;
+use geojson::Geometry;
+use snafu::ensure;
+
+pub struct MultiPoint<'g> {
+    coordinates: &'g [Coordinate2D],
```
Comment (jdroenner) [inline datatypes/src/primitives/multi_point.rs]: to be more precise this field could be named "point_coordinates". Additionally, we should think of a way to remove the lifetime in the future. Maybe we could do something like T: AsRef<[Coordinate2D]>. Or maybe this is a MultiPointRef -_-

File: datatypes/src/primitives/feature_data.rs [hunk 3]
```diff
@@ -53,25 +53,101 @@ pub enum FeatureDataRef<'f> {
     NullableCategorical(NullableCategoricalDataRef<'f>),
 }
 
+impl<'f> FeatureDataRef<'f> {
+    pub fn json_values(&self) -> Box<dyn Iterator<Item = serde_json::Value> + '_> {
```
Comment (jdroenner) [inline datatypes/src/primitives/feature_data.rs]: a way to replace the Box<dyn ... would be nice but i think this is fine for the moment.

Comment (ChristianBeilschmidt) [reply datatypes/src/primitives/feature_data.rs]: I can't think of an easy way to not use a vtable right now. If you have a proposal, I will try to realize it.

File: datatypes/src/collections/multi_point_collection.rs [hunk 4]
```diff
@@ -586,6 +586,205 @@ impl FeatureCollection for MultiPointCollection {
...
+        }
+
+        // creates a foreign member object out of a *when* event value
+        fn foreign_memberize(
+            when_value: serde_json::Value,
+        ) -> serde_json::Map<String, serde_json::Value> {
+            let mut map = serde_json::Map::with_capacity(1);
+            map.insert("when".to_string(), when_value);
+            map
+        };
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply datatypes/src/collections/multi_point_collection.rs]: 467f96d

File: datatypes/src/collections/feature_collection.rs [hunk 5]
```diff
@@ -40,6 +40,17 @@ pub trait FeatureCollection {
     fn remove_column(&self, column: &str) -> Result<Self>
     where
         Self: Sized;
+
+    /// Serialize the feature collection to a geo json string
+    fn to_geo_json(&self) -> String;
+}
+
+pub trait HasGeometryIterator {
```
[the remark this answers is not in the document]
Comment (ChristianBeilschmidt) [reply datatypes/src/collections/feature_collection.rs]: c35e159

File: datatypes/src/primitives/multi_point.rs (hunk 2 above)
Comment (ChristianBeilschmidt) [reply datatypes/src/primitives/multi_point.rs]: dc6e723

**Pull request merged**